File: src/main/java/com/marketplace/demo/controller/dto/PostDTO.java
package com.marketplace.demo.controller.dto;

import java.util.List;

public record PostDTO(Long id, String name, String description,
                      Long views, Long product, Long user,
                      List<Long> likedUsers, List<Long> tags,
                      List<Long> images, List<Long> reviews) {}

File: recsys/recsysMicroservice/Domain/PostDTO.py
from typing import List

from pydantic import BaseModel


class PostDTO(BaseModel):
    id: int
    name: str
    description: str
    products_in_post: List[int]
    user: int
    liked_users: List[int]
    tags: List[int]
    images: List[int]
